src/test/scala/com/microsoft/hyperspace/goldstandard/PlanStabilitySuite.scala

```diff
@@ -0,0 +1,292 @@
+ *
+ * The below license was copied from: https://github.com/FasterXML/jackson-module-scala/blob/2.10/src/main/resources/META-INF/LICENSE
+ */
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.microsoft.hyperspace.goldstandard
+
+import java.io.File
+import java.nio.charset.StandardCharsets
+
+import scala.collection.mutable
+
+import org.apache.commons.io.FileUtils
+import org.apache.spark.internal.Logging
+import org.apache.spark.sql.catalyst.expressions.AttributeSet
+import org.apache.spark.sql.catalyst.util._
+import org.apache.spark.sql.execution._
+import org.apache.spark.sql.execution.command.ExplainCommand
+import org.apache.spark.sql.execution.exchange.{Exchange, ReusedExchangeExec}
+
+// scalastyle:off filelinelengthchecker
+/**
+ * Check that TPC-DS SparkPlans don't change.
+ * If there are plan differences, the error message looks like this:
+ *   Plans did not match:
+ *   last approved simplified plan: /path/to/tpcds-plan-stability/approved-plans-xxx/q1/simplified.txt
+ *   last approved explain plan: /path/to/tpcds-plan-stability/approved-plans-xxx/q1/explain.txt
+ *   [last approved simplified plan]
+ *
+ *   actual simplified plan: /path/to/tmp/q1.actual.simplified.txt
+ *   actual explain plan: /path/to/tmp/q1.actual.explain.txt
+ *   [actual simplified plan]
+ *
+ * The explain files are saved to help debug later, they are not checked. Only the simplified
+ * plans are checked (by string comparison).
+ *
+ *
+ * To run the entire test suite:
+ * {{{
+ *   sbt "test:testOnly *PlanStabilitySuite"
+ * }}}
+ *
+ * To run a single test file upon change:
+ * {{{
+ *  sbt "test:testOnly *PlanStabilitySuite -- -z (tpcds-v1.4/q49)"
+ * }}}
+ *
+ * To re-generate golden files for entire suite, run:
+ * {{{
+ *   SPARK_GENERATE_GOLDEN_FILES=1 sbt "test:testOnly *PlanStabilitySuite"
+ * }}}
+ *
+ * To re-generate golden file for a single test, run:
+ * {{{
+ *   SPARK_GENERATE_GOLDEN_FILES=1 sbt "test:testOnly *PlanStabilitySuite -- -z (tpcds-v1.4/q49)"
+ * }}}
+ */
+// scalastyle:on filelinelengthchecker
+
+trait PlanStabilitySuite extends TPCDSBase with Logging {
+
+  private val originalCrossJoinEnabled = spark.conf.get("spark.sql.crossJoin.enabled")
```

Comment:
Does it make sense to use `withSQLConf` instead?

For example:
```scala
// Enable cross join because some queries fail during query optimization phase.
withSQLConf("spark.sql.crossJoin.enabled" -> "true") {
  tpcdsQueries.foreach { q =>
    test(s"check simplified (tpcds-v1.4/$q)") {
      testQuery("tpcds/queries", q)
    }
  }
}
```
If this doesn't work, the current approach seems fine.

Reply:
thanks @imback82 , i made this change. This is good for localizing the configs to the tests. In future PRs, all such confs will come from a config(properties file) to make things easier but that's for later.